pkg/handler/project/project.go

```diff
@@ -1391,44 +1391,107 @@ func (h *handler) CreateWorkUnit(c *gin.Context) {
-				EmployeeID:      "f32d08ca-8863-4ab3-8c84-a11849451eb8",
-				Name:            "Nguyễn Ngô Lập",
-				Avatar:          "https://s3-ap-southeast-1.amazonaws.com/fortress-images/2870969541970972723.png",
-			},
-		},
-		Stacks: []view.MetaData{
-			{
-				ID:   "f32d08ca-8863-4ab3-8c84-a11849451eb7",
-				Code: "golang",
-				Name: "Golang",
-			},
-			{
-				ID:   "f32d08ca-8863-4ab3-8c84-a11849451eb8",
-				Code: "gcloud",
-				Name: "Google Cloud",
-			},
-		},
-		Type:      "Repository",
-		Status:    "Active",
-		ProjectID: "f32d08ca-8863-4ab3-8c84-a11849451eb7",
+	exists, err := h.store.Project.IsExist(h.repo.DB(), input.ProjectID)
+	if err != nil {
+		l.Error(err, "failed to check project existence")
+		c.JSON(http.StatusInternalServerError, view.CreateResponse[any](nil, nil, err, input, ""))
+		return
+	}
+
+	if !exists {
+		l.Error(ErrProjectNotFound, "project not found")
+		c.JSON(http.StatusNotFound, view.CreateResponse[any](nil, nil, ErrProjectNotFound, nil, ""))
+		return
+	}
+
+	tx, done := h.repo.NewTransaction()
+
+	workUnit := &model.WorkUnit{
+		Name:      input.Body.Name,
+		Type:      model.WorkUnitType(input.Body.Type),
+		Status:    model.WorkUnitStatus(input.Body.Status),
+		SourceURL: input.Body.URL,
+		ProjectID: model.MustGetUUIDFromString(input.ProjectID),
+	}
+
+	if err := h.store.WorkUnit.Create(tx.DB(), workUnit); err != nil {
+		l.Error(err, "failed to create new work unit")
+		c.JSON(http.StatusInternalServerError, view.CreateResponse[any](nil, nil, done(err), input, ""))
+		return
+	}
+
+	// create work unit stack
+	for _, stackID := range input.Body.Stacks {
+		stack, err := h.store.Stack.One(tx.DB(), stackID)
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			l.Error(err, "stack not found")
+			c.JSON(http.StatusBadRequest, view.CreateResponse[any](nil, nil, done(ErrStackNotFound), input, ""))
+			return
+		}
+		if err != nil && !errors.Is(err, gorm.ErrRecordNotFound) {
+			l.Error(err, "failed to get stacks by ID")
+			c.JSON(http.StatusInternalServerError, view.CreateResponse[any](nil, nil, done(err), input, ""))
+			return
+		}
+
+		wuStack := model.WorkUnitStack{
+			StackID:    model.MustGetUUIDFromString(stackID),
+			WorkUnitID: workUnit.ID,
+		}
+		if err := h.store.WorkUnitStack.Create(tx.DB(), &wuStack); err != nil {
+			l.Error(err, "failed to create new work unit stack")
+			c.JSON(http.StatusInternalServerError, view.CreateResponse[any](nil, nil, done(err), input, ""))
+			return
+		}
+
+		wuStack.Stack = *stack
+		workUnit.WorkUnitStacks = append(workUnit.WorkUnitStacks, wuStack)
+	}
+
+	// create work unit member
+	for _, employeeID := range input.Body.Members {
+		employee, err := h.store.Employee.One(tx.DB(), employeeID)
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			l.Error(err, "stack not found")
+			c.JSON(http.StatusBadRequest, view.CreateResponse[any](nil, nil, done(ErrEmployeeNotFound), input, ""))
+			return
+		}
+		if err != nil && !errors.Is(err, gorm.ErrRecordNotFound) {
```

Comment:
we dont need to check error != gorm.ErrRecordNotFound here, because when it come here, the condition always true